src/main/java/pl/psnc/dei/service/context/ContextService.java

```diff
@@ -0,0 +1,15 @@
+package pl.psnc.dei.service.context;
+
+import pl.psnc.dei.model.Record;
+
+/**
+ * Service used for manipulation on concrete service
+ * @param <T> context class being served by this service
+ */
+public abstract class ContextService<T> {
```

Comment:
Interface would be better than an abstract class

Reply:
moved to be interface